runtime/studentAPI.py

```diff
@@ -280,6 +281,76 @@ def hibike_write_value(self, uid, params):
         """Writes parameters to ``uid``."""
         self.to_manager.put([HIBIKE_COMMANDS.WRITE, [uid, params]])
 
-    def get_gamecode(self):
+    def _get_gamecodes(self):
         """Get a gamecode."""
-        return self._get_sm_value("gamecode")
+        return self._get_sm_value("gamecodes")
+
+    def _get_gamecodes_check(self):
+        """Get gamecode answers"""
+        return self._get_sm_value("gamecodes_check")
+
+    def decode_message(self, rfid_seed):
+        """Method for 2018 Game: Solar Scramble
+
+        This method will use the students functions to decode a message,
+        and display the solution to Dawn
+
+        This function takes in a RFID tag and returns
+        True on success, False on error"""
+        def identity(value):
+            '''
+            Used only in the (hopefully) rare event that none of the other
+            functions are bijections with a given domain of RFIDs
+            '''
+            return value
+
+        def limit_input_to(limit):
+            '''Generate a function to limit size of inputs'''
+            def retval(input_val):
+                while input_val > limit:
+                    input_val = (input_val % limit) + (input_val // limit)
+                return input_val
+            return retval
+
+        def compose_funcs(func_a, func_b):
+            '''
+            Composes two single-input functions together, A(B(x))
+            '''
+            def composed(input_x):
+                return func_a(func_b(input_x))
+            return composed
+        try:
+            index = self._get_sm_value("rfids").index(rfid_seed)
+            challenge_code = self._get_gamecodes()[index]
+            check_challenge_code = self._get_gamecodes_check()[index]
+        except ValueError:
+            return False
+
+        func_map = {}
+        func_map[0] = identity
+        func_map[1] = next_power
+        func_map[2] = reverse_digits
+        func_map[3] = compose_funcs(smallest_prime_fact, limit_input_to(1000000))
+        func_map[4] = double_caesar_cipher
+        func_map[5] = silly_base_two
+        func_map[6] = most_common_digit
+        func_map[7] = valid_isbn_ten
+        func_map[8] = simd_four_square
```

Comment:
This could just use a normal dict literal; i.e.
```py
func_map = {
    0: identity,
    1: next_power,
    # etc.
}
```

Reply:
Better yet, an array, since these are integer keys